### PR TITLE
Pairtools sample

### PR DESCRIPTION
**Pull request by jianhong**
## PR checklist

Closes #XXX 

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/pairtools/sample/main.nf [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    output:
+    tuple val(meta), path("*.pairsam.gz")  , emit: pairsam
+    path "versions.yml"                    , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}_subsample"
```
Comment (nvnieuwk) [inline modules/nf-core/pairtools/sample/main.nf:22]: It's preffered to keep the default of the `prefix` as `meta.id`. The guidelines state that for modules that can have the same input and output, you should add a check with an error message: see [here](https://nf-co.re/docs/contributing/modules#naming-conventions) paragraph 6

File: modules/nf-core/pairtools/sample/main.nf [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    tag "$meta.id"
+    label 'process_high'
+
+    conda "bioconda::pairtools=0.3.0"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/pairtools:0.3.0--py37hb9c2fc3_5' :
+        'quay.io/biocontainers/pairtools:0.3.0--py37hb9c2fc3_5' }"
+
+    input:
+    tuple val(meta), path(input), val(fraction)
```
Comment (nvnieuwk) [inline modules/nf-core/pairtools/sample/main.nf:11]: Could you change the `input` variable name to something a bit more descriptive (like file type, file content...)

File: tests/modules/nf-core/pairtools/sample/main.nf [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+nextflow.enable.dsl = 2
+
+include { PAIRTOOLS_SAMPLE } from '../../../../../modules/nf-core/pairtools/sample/main.nf'
+
+workflow test_pairtools_sample {
+
+    input = [
+        [ id:'test', single_end:false ], // meta map
+        file('https://raw.githubusercontent.com/open2c/pairtools/master/tests/data/mock.4dedup.pairsam', checkIfExists: true),
```
Comment (nvnieuwk) [inline tests/modules/nf-core/pairtools/sample/main.nf:11]: Can this file be added to the test-dataset repository?

**Review by nvnieuwk — CHANGES_REQUESTED** [review]
Hi, good job on this module! Here are some suggestions to improve the module even more :)

File: modules/nf-core/pairtools/sample/meta.yml [hunk 4]
```diff
@@ -0,0 +1,44 @@
+name: "pairtools_sample"
+description: Select a random subset of pairs in a pairs file.
+keywords:
+  - sort
```
Comment (nvnieuwk) [inline modules/nf-core/pairtools/sample/meta.yml:4]: Can you add some more keywords here?

Comment (jianhong) [reply modules/nf-core/pairtools/sample/meta.yml]: sorry, my fault.

Comment (nvnieuwk) [reply modules/nf-core/pairtools/sample/meta.yml]: No problem :) That's what reviews are for 😁

**Comment by jianhong** [comment]
@nservant Could you please take a look about this module?

**Pull request closed without merging**